## 83% of end-end tests have passed
```
Total executed tests: 77
Failed tests: 9
Expected to fail tests: 4
Passed tests: 64
Not executed tests: 1
```
```
IDs of not executed tests: 742063 
```
```
IDs of failed tests: 703133,727230,741925,703202,741054,702894,739307,702947,742016 
```
```
IDs of expected to fail tests: 702844,741924,703503,740490 
```
<h3>Not executed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=tests/l2-swap&PR_ID=22419&APK_NAME=22419.apk&TR_CASE_IDS=742063">Rerun not executed tests</a></li></details><h3>Failed tests (9)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=tests/l2-swap&PR_ID=22419&APK_NAME=22419.apk&TR_CASE_IDS=703133,727230,741925,703202,741054,702894,739307,702947,742016">Rerun failed tests</a></li><br/><h4>Class TestCommunityMultipleDeviceMergedThree:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827952">test_community_send_message_from_offline</a>, id: 741925 </b></td></tr><tr><td><p><blockquote><div>Device 2: Click until `ChatMessageInput` by `accessibility id`: `chat-message-input` will be presented </div><div>Device 2: Looking for a message by text: message in 1-1 chat </div></blockquote></p>

```
communities/test_community.py:1413: in test_community_send_message_from_offline
    self.errors.verify_no_errors()
base_test_case.py:205: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 1: Can't receive the message 'message community' in community if it's sent from offline
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESrndFKw4Sp6GXnpZE">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESN2FPtAu1fGAi2Zqk">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827916">test_activity_center_reply_read_unread_delete_filter_swipe</a>, id: 702947 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: CommunitiesTab </div><div>Device 2: Looking for a message by text: something to reply to </div></blockquote></p>

```
activity_center/test_activity_center.py:302: in test_activity_center_reply_read_unread_delete_filter_swipe
    self.channel_2.chat_element_by_text(message_to_reply).wait_for_visibility_of_element(120)
../views/base_element.py:139: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: ChatElementByText by xpath:`//*[starts-with(@text,'something to reply to')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESa1IOLslPNCmdkLVf">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESmOycp2DUxRfRoBHJ">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827923">test_group_chat_reactions</a>, id: 703202 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Button` by `accessibility id`: `reaction-sad` </div><div>Device 2: Find `Button` by `accessibility id`: `reaction-sad` </div></blockquote></p>

```
private_chats/test_group_chat.py:173: in test_group_chat_reactions
    self.chats[1].set_reaction(message=message, emoji="sad")
../views/chat_view.py:695: in set_reaction
    element.double_click()
../views/base_element.py:114: in double_click
    [self.find_element().click() for _ in range(2)]
../views/base_element.py:114: in <listcomp>
    [self.find_element().click() for _ in range(2)]
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 2: Button by accessibility id: `reaction-sad` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES6oF1Twxen9yNDAj5">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESqdSmoPJs8EKqezsY">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESwDwt7qO0I8qAiIyu">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestFallbackMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827941">test_fallback_add_key_pair</a>, id: 741054 </b></td></tr><tr><td><p><blockquote><div>Device 1: Swiping left on element Button </div><div>Device 1: Find `Button` by `xpath`: `//android.view.ViewGroup[contains(@content-desc,'Imported account')]` </div></blockquote></p>

```
profile/test_fallback.py:211: in test_fallback_add_key_pair
    wallet_1.get_account_element(account_name=imported_key_pair_account_name).swipe_left_on_element()
../views/base_element.py:296: in swipe_left_on_element
    location, size = self.get_element_coordinates()
../views/base_element.py:289: in get_element_coordinates
    element = self.find_element()
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: Button by xpath: `//android.view.ViewGroup[contains(@content-desc,'Imported account')]` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESpGEWT6dNYju2em6g">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESOaXmmybCdYJ5W6Oe">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES5cTDFWaYmegmZWLl">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827921">test_restore_multiaccount_with_waku_backup_remove_profile_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div># STEP: Check that removed user is not shown in the list anymore</div></blockquote></p>

```
communities/test_community.py:245: in test_restore_multiaccount_with_waku_backup_remove_profile_switch
    self.errors.verify_no_errors()
base_test_case.py:205: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 1: zQ3...dWXh5 was not restored as a contact from waku backup!
E    Device 1: zQ3...Vacac was not restored as a contact from waku backup!
E    Device 1: admin_open was not restored from waku-backup!!
E    Device 1: member_open was not restored from waku-backup!!
E    Device 1: admin_closed was not restored from waku-backup!!
E    Device 1: member_closed was not restored from waku-backup!!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESrI9XfUehDEIBy2wD">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827934">test_wallet_send_erc20_from_drawer[Optimism Sepolia-USD Coin-USDC-2-0.01]</a>, id: 727230 </b></td></tr><tr><td><p><blockquote><div>Device 1: Double tap on: Button </div><div>Device 1: Find `Button` by `xpath`: `//*[@content-desc='send']/*[@content-desc='left-icon-for-action']` </div></blockquote></p>

```
wallet_txs/test_wallet_testnet.py:81: in test_wallet_send_erc20_from_drawer
    self.wallet_1.send_asset_from_drawer(address=self.receiver['wallet_address'],
../views/wallet_view.py:290: in send_asset_from_drawer
    self.send_from_drawer_button.double_click()
../views/base_element.py:114: in double_click
    [self.find_element().click() for _ in range(2)]
../views/base_element.py:114: in <listcomp>
    [self.find_element().click() for _ in range(2)]
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: Button by xpath: `//*[@content-desc='send']/*[@content-desc='left-icon-for-action']` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESKW1Vr5RqH0Fwlgwn">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESjsaRJw7lfaK3ZbSN">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827962">test_wallet_send_erc20_from_drawer[Sepolia-USD Coin-USDC-2-0.01]</a>, id: 742016 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `xpath`: `//*[@content-desc='send']/*[@content-desc='left-icon-for-action']` </div><div>Device 1: Find `Button` by `xpath`: `//*[@content-desc='network-list']/*[@text='Sepolia']` </div></blockquote></p>

```
wallet_txs/test_wallet_testnet.py:81: in test_wallet_send_erc20_from_drawer
    self.wallet_1.send_asset_from_drawer(address=self.receiver['wallet_address'],
../views/wallet_view.py:291: in send_asset_from_drawer
    self.select_network(network_name)
../views/wallet_view.py:258: in select_network
    xpath="//*[@content-desc='network-list']/*[@text='%s']" % network_name).click()
../views/base_element.py:90: in click
    element = self.find_element()
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: Button by xpath: `//*[@content-desc='network-list']/*[@text='Sepolia']` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESKW1Vr5RqH0Fwlgwn">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESjsaRJw7lfaK3ZbSN">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestDeepLinksOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827936">test_deep_links_communities</a>, id: 739307 </b></td></tr><tr><td><p><blockquote><div>Device 1: Waiting 35s for dapp to load </div><div>Device 1: Find `Text` by `accessibility id`: `community-title` </div></blockquote></p>

```
deep_links/test_deep_and_universal_links.py:105: in test_deep_links_communities
    if self.community_view.community_title.text != text:
../views/base_element.py:424: in text
    text = self.find_element().text
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: Text by accessibility id: `community-title` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESGxh9LZ9cgAqz0nBi">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827915">test_community_contact_block_unblock_offline</a>, id: 702894 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Text` by `xpath`: `//*[starts-with(@text,'Hurray! unblocked')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']//*[@content-desc='message-status']/android.widget.TextView` </div><div>Device 2: `Text` is `Sent` </div></blockquote></p>

```
communities/test_community.py:789: in test_community_contact_block_unblock_offline
    self.errors.verify_no_errors()
base_test_case.py:205: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 2: Message was not delivered after back up online.
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8fjRaja2OP3DlU8w">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESC5alOKb6qAz55jFN">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Expected to fail tests (4)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827907">test_community_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 702844 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `EmojisNumber` by `xpath`: `//*[starts-with(@text,'https://m.youtube.com/watch?v=Je7yErjEVt4')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']/../..//*[@content-desc='emoji-reaction-4']/android.widget.TextView[2]` </div><div>Device 2: Element EmojisNumber text is equal to 1 </div></blockquote></p>

```
communities/test_community.py:665: in test_community_links_with_previews_github_youtube_twitter_gif_send_enable
    self.errors.verify_no_errors()
base_test_case.py:205: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 1: No preview is loaded for url https://youtu.be/Je7yErjEVt4
E    Device 1: No preview is loaded for url https://www.youtube.com/watch?v=XN-SVmuJH2g&list=PLbrz7IuP1hrgNtYe9g6YHwHO6F3OqNMao
E    Device 1: No preview is loaded for url https://m.youtube.com/watch?v=Je7yErjEVt4 
```

<b>[[Youtube links preview is not loaded on LambdaTest emulators, needs investigation]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8fjRaja2OP3DlU8w">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESC5alOKb6qAz55jFN">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827940">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `network-dropdown` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
wallet/test_wallet_mainnet_no_send.py:71: in test_wallet_balance_mainnet
    self.errors.verify_no_errors()
base_test_case.py:205: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 1: For the USD Coin the wrong value 0.0 is shown, expected 5.0 in total
E    Device 1: For the USD Coin the wrong value 0.0 is shown, expected 5.0 on Optimism 
```

<b>[[Might fail due to #22384]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESnTDb3PsOgs048PDo">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMergedThree:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827951">test_community_mobile_data_and_wi_fi_only_fetching</a>, id: 741924 </b></td></tr><tr><td><p><blockquote><div>Device 2: Click until `ChatMessageInput` by `accessibility id`: `chat-message-input` will be presented </div><div>Device 2: Looking for a message by text: message text 2 </div></blockquote></p>

```
communities/test_community.py:1359: in test_community_mobile_data_and_wi_fi_only_fetching
    self.errors.verify_no_errors()
base_test_case.py:205: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 2: Message 'message text 2' in community, which is sent when receiver was offline, is received using mobile data 
```

<b>[[Community message can be fetched from offline when using mobile data]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESrndFKw4Sp6GXnpZE">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESN2FPtAu1fGAi2Zqk">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827929">test_community_discovery</a>, id: 703503 </b></td></tr><tr><td><p><blockquote><div>Device 1: Image differs from template to 0.0 percents </div><div>Device 1: Find `BaseElement` by `xpath`: `//*[@content-desc='community-title']/preceding-sibling::*/android.widget.ImageView` </div></blockquote></p>

```
communities/test_community.py:294: in test_community_discovery
    if self.community_view.community_logo.is_element_differs_from_template('status_community_logo.png'):
../views/base_element.py:267: in is_element_differs_from_template
    difference = ImageChops.difference(self.image, self.template)
../views/base_element.py:245: in image
    return Image.open(BytesIO(base64.b64decode(self.find_element().screenshot_as_base64)))
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: BaseElement by xpath: `//*[@content-desc='community-title']/preceding-sibling::*/android.widget.ImageView` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception 
```

<b>[[Might fail if the Discover curation dapp is too slow to load > 2 mins]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESrI9XfUehDEIBy2wD">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (64)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827898">test_group_chat_join_send_text_messages_push</a>, id: 702807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES6oF1Twxen9yNDAj5">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESqdSmoPJs8EKqezsY">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESwDwt7qO0I8qAiIyu">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827927">test_group_chat_mute_chat</a>, id: 703495 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES6oF1Twxen9yNDAj5">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESqdSmoPJs8EKqezsY">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESwDwt7qO0I8qAiIyu">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827888">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES6oF1Twxen9yNDAj5">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESqdSmoPJs8EKqezsY">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESwDwt7qO0I8qAiIyu">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827899">test_group_chat_offline_pn</a>, id: 702808 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES6oF1Twxen9yNDAj5">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESqdSmoPJs8EKqezsY">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESwDwt7qO0I8qAiIyu">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827924">test_group_chat_send_image_save_and_share</a>, id: 703297 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES6oF1Twxen9yNDAj5">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESqdSmoPJs8EKqezsY">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESwDwt7qO0I8qAiIyu">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePRTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827919">test_activity_center_admin_notification_accept_swipe</a>, id: 702958 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESiAATPF0dykUcemNR">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES7OD5jRovDgo95CDM">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827918">test_activity_center_mentions</a>, id: 702957 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESiAATPF0dykUcemNR">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES7OD5jRovDgo95CDM">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827947">test_activity_centre_contact_request_receiver_offline</a>, id: 741809 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESiAATPF0dykUcemNR">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES7OD5jRovDgo95CDM">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDeviceTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827935">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESSqOrGZ8MFRH5fcmd">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestAndroid13:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827946">test_create_account_android_13</a>, id: 741807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESgDkkEGFsKeucEj0T">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827890">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESrI9XfUehDEIBy2wD">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827925">test_community_mute_community_and_channel</a>, id: 703382 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESrI9XfUehDEIBy2wD">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827914">test_community_undo_delete_message</a>, id: 702869 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESrI9XfUehDEIBy2wD">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827909">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESrI9XfUehDEIBy2wD">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827920">test_community_mark_all_messages_as_read</a>, id: 703086 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8fjRaja2OP3DlU8w">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESC5alOKb6qAz55jFN">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827902">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8fjRaja2OP3DlU8w">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESC5alOKb6qAz55jFN">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827906">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8fjRaja2OP3DlU8w">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESC5alOKb6qAz55jFN">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827922">test_community_several_images_send_reply</a>, id: 703194 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8fjRaja2OP3DlU8w">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESC5alOKb6qAz55jFN">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827903">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8fjRaja2OP3DlU8w">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESC5alOKb6qAz55jFN">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827905">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8fjRaja2OP3DlU8w">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESC5alOKb6qAz55jFN">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827913">test_community_one_image_send_reply_set_reaction</a>, id: 702859 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8fjRaja2OP3DlU8w">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESC5alOKb6qAz55jFN">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827932">test_community_edit_delete_message_when_offline</a>, id: 704615 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8fjRaja2OP3DlU8w">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESC5alOKb6qAz55jFN">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827904">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES8fjRaja2OP3DlU8w">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESC5alOKb6qAz55jFN">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUiTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827928">test_1_1_chat_mute_chat</a>, id: 703496 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESQUtpooVeXUY59BHw">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESrZOVrR91qkvnCcte">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827895">test_1_1_chat_is_shown_message_sent_delivered_from_offline</a>, id: 702783 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESQUtpooVeXUY59BHw">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESrZOVrR91qkvnCcte">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827896">test_1_1_chat_delete_via_long_press_relogin</a>, id: 702784 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESQUtpooVeXUY59BHw">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESrZOVrR91qkvnCcte">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestProfileOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827954">test_profile_back_up_seed_phrase_validation</a>, id: 741966 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESitqWfg1ppwDUj0c9">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827955">test_profile_change_currency</a>, id: 741967 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESitqWfg1ppwDUj0c9">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestFallbackMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827938">test_fallback_with_correct_seed_phrase</a>, id: 740221 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESpGEWT6dNYju2em6g">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESOaXmmybCdYJ5W6Oe">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES5cTDFWaYmegmZWLl">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827937">test_fallback_sync_with_error</a>, id: 740220 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESpGEWT6dNYju2em6g">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESOaXmmybCdYJ5W6Oe">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES5cTDFWaYmegmZWLl">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827939">test_fallback_validate_seed_phrase</a>, id: 740222 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESpGEWT6dNYju2em6g">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESOaXmmybCdYJ5W6Oe">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES5cTDFWaYmegmZWLl">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMergedTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827897">test_community_mentions_push_notification</a>, id: 702786 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES2LcLhIeVxgaFSfgc">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESMiOMVmRYLhhuOXvw">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827930">test_community_join_when_node_owner_offline</a>, id: 703629 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES2LcLhIeVxgaFSfgc">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESMiOMVmRYLhhuOXvw">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827917">test_community_hashtag_links_to_community_channels</a>, id: 702948 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES2LcLhIeVxgaFSfgc">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESMiOMVmRYLhhuOXvw">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827900">test_community_markdown_support</a>, id: 702809 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES2LcLhIeVxgaFSfgc">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESMiOMVmRYLhhuOXvw">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827908">test_community_leave</a>, id: 702845 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES2LcLhIeVxgaFSfgc">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESMiOMVmRYLhhuOXvw">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletCollectibles:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827949">test_wallet_send_collectible</a>, id: 741840 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESItoHNTMIhRwmmUJE">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827948">test_wallet_collectibles_balance</a>, id: 741839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESItoHNTMIhRwmmUJE">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827950">test_wallet_collectible_send_from_expanded_info_view</a>, id: 741841 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESItoHNTMIhRwmmUJE">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827942">test_wallet_send_flow_mainnet</a>, id: 741554 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESnTDb3PsOgs048PDo">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827944">test_wallet_bridge_flow_mainnet</a>, id: 741612 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESnTDb3PsOgs048PDo">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestAndroid12:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827945">test_create_account_android_12</a>, id: 741806 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESQDjcgDxDSU2ypYDL">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityCenterContactRequestMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827910">test_activity_center_contact_request_decline</a>, id: 702850 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESolRQeSEMkbi8FLnn">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESSkU6VbrJlFB9a4GA">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827893">test_add_contact_field_validation</a>, id: 702777 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESolRQeSEMkbi8FLnn">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESSkU6VbrJlFB9a4GA">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827911">test_activity_center_contact_request_accept_swipe_mark_all_as_read</a>, id: 702851 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESolRQeSEMkbi8FLnn">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESSkU6VbrJlFB9a4GA">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMergedThree:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827953">test_community_messaging_on_mobile_data</a>, id: 741926 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESrndFKw4Sp6GXnpZE">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESN2FPtAu1fGAi2Zqk">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDeviceThree:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827943">test_wallet_swap_flow_mainnet</a>, id: 741555 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESSaRG7v7GhVfZpFgf">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestProfileMultipleDevices:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827960">test_profile_allow_new_contact_requests_toggle</a>, id: 741972 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESmQHPcbrB1kHjFRRi">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESq8VZfKXYXYypfKs0">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827959">test_profile_change_accent_color</a>, id: 741971 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESmQHPcbrB1kHjFRRi">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESq8VZfKXYXYypfKs0">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827956">test_profile_change_username</a>, id: 741968 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESmQHPcbrB1kHjFRRi">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESq8VZfKXYXYypfKs0">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827958">test_profile_set_bio</a>, id: 741970 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESmQHPcbrB1kHjFRRi">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESq8VZfKXYXYypfKs0">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827957">test_profile_change_profile_photo</a>, id: 741969 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESmQHPcbrB1kHjFRRi">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESq8VZfKXYXYypfKs0">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827894">test_1_1_chat_emoji_send_reply_and_open_link</a>, id: 702782 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESfMSwUQih6cn8gjR2">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESyQCSCMtP9bRkIPIb">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827891">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESfMSwUQih6cn8gjR2">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESyQCSCMtP9bRkIPIb">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827901">test_1_1_chat_push_emoji</a>, id: 702813 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESfMSwUQih6cn8gjR2">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESyQCSCMtP9bRkIPIb">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827886">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESfMSwUQih6cn8gjR2">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESyQCSCMtP9bRkIPIb">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827889">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESfMSwUQih6cn8gjR2">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESyQCSCMtP9bRkIPIb">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827912">test_1_1_chat_edit_message</a>, id: 702855 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESfMSwUQih6cn8gjR2">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESyQCSCMtP9bRkIPIb">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827887">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESfMSwUQih6cn8gjR2">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESyQCSCMtP9bRkIPIb">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827926">test_1_1_chat_send_image_save_and_share</a>, id: 703391 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESfMSwUQih6cn8gjR2">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESyQCSCMtP9bRkIPIb">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827961">test_send_eth[Arbitrum Sepolia-0.0001]</a>, id: 742015 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESKW1Vr5RqH0Fwlgwn">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESjsaRJw7lfaK3ZbSN">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827933">test_send_eth[Status Network Sepolia-0.0002]</a>, id: 727229 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESKW1Vr5RqH0Fwlgwn">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESjsaRJw7lfaK3ZbSN">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestDeepLinksOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827892">test_links_deep_links_profile</a>, id: 702775 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESGxh9LZ9cgAqz0nBi">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3827931">test_links_open_universal_links_from_chat</a>, id: 704613 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESGxh9LZ9cgAqz0nBi">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/test/appium/support/testrail_report.py
+++ b/test/appium/support/testrail_report.py
@@ -113,6 +113,7 @@ class TestrailReport(BaseTestReport):
         test_categories['fallback'] = 63472
         test_categories['android_versions'] = 63809
         test_categories['profile'] = 63854
+        test_categories['real_tx'] = 63867
 
         case_ids = list()
         for arg in argv:

--- a/test/appium/tests/wallet/test_wallet_mainnet_no_send.py
+++ b/test/appium/tests/wallet/test_wallet_mainnet_no_send.py
@@ -19,12 +19,12 @@ class TestWalletOneDevice(MultipleSharedDeviceTestCase):
         self.drivers, self.loop = create_shared_drivers(1)
         self.sign_in_view = SignInView(self.drivers[0])
         self.sender, self.receiver = transaction_senders['ETH_1'], transaction_senders['ETH_2']
-        self.total_balance = {'Ether': 0.0362, 'USD Coin': 5.0, 'Status': 13.0, 'Uniswap': 0.627, 'Dai Stablecoin': 0.0}
-        self.mainnet_balance = {'Ether': 0.015, 'USD Coin': 0.0, 'Status': 10.0, 'Uniswap': 0.127,
+        self.total_balance = {'Ether': 0.0362, 'Status': 13.0, 'Uniswap': 0.627, 'Dai Stablecoin': 0.0}
+        self.mainnet_balance = {'Ether': 0.015,  'Status': 10.0, 'Uniswap': 0.127,
                                 'Dai Stablecoin': 0.0}
-        self.optimism_balance = {'Ether': 0.0011, 'USD Coin': 5.0, 'Status': 3.0, 'Uniswap': 0, 'Dai Stablecoin': 0.0}
-        self.arb_balance = {'Ether': 0.0051, 'USD Coin': 0.0, 'Status': 0.0, 'Uniswap': 0.5, 'Dai Stablecoin': 0.0}
-        self.base_balance = {'Ether': 0.015, 'USD Coin': 0.0, 'Status': 0.0, 'Uniswap': 0.0, 'Dai Stablecoin': 0.0}
+        self.optimism_balance = {'Ether': 0.0011,  'Status': 3.0, 'Uniswap': 0, 'Dai Stablecoin': 0.0}
+        self.arb_balance = {'Ether': 0.0051,'Status': 0.0, 'Uniswap': 0.5, 'Dai Stablecoin': 0.0}
+        self.base_balance = {'Ether': 0.015, 'Status': 0.0, 'Uniswap': 0.0, 'Dai Stablecoin': 0.0}
         self.sender['wallet_address'] = '0x' + self.sender['address']
         self.receiver['wallet_address'] = '0x' + self.receiver['address']
         self.sign_in_view.recover_access(passphrase=self.sender['passphrase'])
@@ -37,8 +37,7 @@ class TestWalletOneDevice(MultipleSharedDeviceTestCase):
         self.wallet_view = self.home_view.wallet_tab.click()
         self.account_name = 'Account 1'
 
-    @marks.testrail_id(740490)
-    @marks.xfail(reason="Might fail due to #22384")
+    @marks.testrail_id(740490) #TODO: add USDc check back when token collision USDc and USDCN is resolved
     def test_wallet_balance_mainnet(self):
         self.wallet_view.just_fyi("Checking total balance")
         real_balance = {}

--- a/test/appium/tests/wallet_txs/test_wallet_mainnet_txs.py
+++ b/test/appium/tests/wallet_txs/test_wallet_mainnet_txs.py
@@ -1,0 +1,72 @@
+import pytest
+
+from base_test_case import MultipleSharedDeviceTestCase, create_shared_drivers
+from views.wallet_view import WalletView
+from tests import marks
+from users import transaction_senders
+from views.sign_in_view import SignInView
+
+
+@pytest.mark.xdist_group(name="three_1")
+@marks.nightly
+class TestWalletOneDeviceThree(MultipleSharedDeviceTestCase):
+
+    def prepare_devices(self):
+        self.drivers, self.loop = create_shared_drivers(1)
+        self.sign_in = SignInView(self.drivers[0])
+        self.sender, self.receiver = transaction_senders['ETH_7'], transaction_senders['ETH_2']
+        self.sign_in.recover_access(passphrase=self.sender['passphrase'])
+
+        self.home_view = self.sign_in.get_home_view()
+        self.sender_username = self.home_view.get_username()
+        self.profile = self.home_view.profile_button.click()
+        self.profile.switch_network()
+        self.sign_in.sign_in(user_name=self.sender_username)
+        self.wallet = self.home_view.wallet_tab.click()
+        self.account_name = 'Account 1'
+
+    @marks.testrail_id(742063)
+    def test_wallet_swap_dai_snt_real_tx(self):
+        self.wallet = WalletView(self.drivers[0])
+        network_name = 'Arbitrum'
+        amount, asset_name_from, from_ticker = 0.01, 'Dai Stablecoin', 'DAI'
+        asset_name_to, to_ticker, decimals_to = 'Status', 'SNT', 18
+
+        fiat_initial_amount_from = self.wallet.get_balance(asset_name_from, fiat=True)
+        fiat_initial_amount_to = self.wallet.get_balance(asset_name_to, fiat=True)
+        fiat_expected_amount_from = self.wallet.round_amount_float(fiat_initial_amount_from - amount, 2)
+        fiat_expected_amount_to = self.wallet.round_amount_float(fiat_initial_amount_to + amount, 2)
+
+        self.home_view.wallet_tab.click()
+        self.wallet.get_account_element().wait_for_rendering_ended_and_click()
+        self.swap_data = self.wallet.swap_asset_from_drawer(asset_name=asset_name_from,
+                                                            amount=amount,
+                                                            network_name=network_name,
+                                                            decimals_to=decimals_to,
+                                                            asset_to=asset_name_to)
+        for key, value in self.swap_data.items():
+            self.wallet.just_fyi(f"Param tx {key} from review page: {value}")
+
+        self.wallet.just_fyi("Check swapping toast")
+        swapping_toast = f"Swapping {amount} {from_ticker} to "
+
+        if not self.wallet.element_by_text_part(swapping_toast).is_element_displayed(10):
+            self.errors.append(self.wallet, f"Can't find {swapping_toast} toast")
+
+        self.wallet.just_fyi("Verify swap tx in the list")
+        amount_in_tx = str(self.wallet.round_amount_float(self.swap_data['receive_amount'], 6))
+        tx_errors = self.wallet.check_last_transaction_in_activity(self.wallet.driver.device_time,
+                                                                   amount,
+                                                                   tx_type='Swap',
+                                                                   asset=from_ticker,
+                                                                   from_account_name=self.account_name,
+                                                                   swap_amount_to=amount_in_tx,
+                                                                   swap_asset_to=to_ticker,
+                                                                   network=network_name)
+        self.errors.append(self.wallet, tx_errors)
+
+        self.wallet.just_fyi("Check that balance is updated for receiver")
+        self.wallet.wait_for_wallet_balance_to_update(fiat_expected_amount_from, asset_name_from, fiat=True)
+        self.wallet.wait_for_wallet_balance_to_update(fiat_expected_amount_to, asset_name_to, fiat=True)
+
+        self.errors.verify_no_errors()

--- a/test/appium/tests/wallet_txs/test_wallet_testnet.py
+++ b/test/appium/tests/wallet_txs/test_wallet_testnet.py
@@ -50,7 +50,7 @@ class TestWalletMultipleDevice(MultipleSharedDeviceTestCase):
 
         self.wallet_1.just_fyi("Verify send tx in the list for sender")
         tx_errors = self.wallet_1.check_last_transaction_in_activity(device_time_before_sending, amount,
-                                                                     to_account=self.receiver['wallet_address'],
+                                                                     send_to_account=self.receiver['wallet_address'],
                                                                      asset=asset_ticker,
                                                                      tx_type='Send',
                                                                      network=network)
@@ -85,7 +85,7 @@ class TestWalletMultipleDevice(MultipleSharedDeviceTestCase):
 
         self.wallet_1.just_fyi("Verify send tx in the list for sender")
         tx_errors = self.wallet_1.check_last_transaction_in_activity(device_time_before_sending, amount,
-                                                                     to_account=self.receiver['wallet_address'],
+                                                                     send_to_account=self.receiver['wallet_address'],
                                                                      asset=asset_ticker,
                                                                      tx_type='Send',
                                                                      network=network)

--- a/test/appium/views/wallet_view.py
+++ b/test/appium/views/wallet_view.py
@@ -13,24 +13,53 @@ from views.sign_in_view import SignInView
 
 
 class AssetElement(Button):
-
     def __init__(self, driver, asset_name):
         self.asset_name = asset_name
-        self.locator = "//android.view.ViewGroup[@content-desc='container']/android.widget.TextView[@text='%s']" % \
-                       self.asset_name
+        self.locator = (
+                "//android.view.ViewGroup[@content-desc='container']/android.widget.TextView[@text='%s']"
+                % self.asset_name
+        )
         super().__init__(driver=driver, xpath=self.locator)
 
+    def _parse_amount(self, element) -> float:
+        """
+        Helper method to parse and return the numeric amount from a given element.
+        Handles cases where the amount starts with a currency symbol or '<'.
+        """
+        try:
+            # Extract the first part of the text
+            amount = element.text.split()[0]
+
+            # Handle cases where the amount starts with '<'
+            if '<' in amount:
+                return 0.01 if '0.01' in amount else 0.0  # Covers cases like '<$0.01'
+
+            # Strip any leading non-numeric characters (e.g., currency symbols)
+            numeric_part = ''.join(c for c in amount if c.isdigit() or c == '.')
+
+            # Convert the remaining string to float
+            return float(numeric_part)
+        except (ValueError, IndexError):
+            # Fail the test if parsing fails
+            pytest.fail(f"Cannot parse amount for {self.asset_name}")
+            return 0.0  # Optional fallback
+
     def get_amount(self):
+        """
+        Get the crypto amount for the asset.
+        """
         element = Text(self.driver, xpath=self.locator + "/../android.widget.TextView[3]")
         element.scroll_to_element()
-        try:
-            amount = element.text.split()[0]
-            if '<' in amount:
-                return 0
-            else:
-                return float(amount)
-        except ValueError:
-            pytest.fail("Cannot get %s amount" % self.asset_name)
+        return self._parse_amount(element)
+
+    def get_fiat_balance(self):
+        """
+        Get the fiat balance for the asset.
+        """
+        element = Text(self.driver, xpath=self.locator + "/../android.widget.TextView[2]")
+        element.scroll_to_element()
+        amount = element.text.split()[1:]
+        return self._parse_amount(element)
 
 
 class CollectibleItemElement(Button):
@@ -70,12 +99,17 @@ class ActivityElement(BaseElement):
                     xpath="//*[@content-desc='context-tag'][1]/android.widget.TextView").text
 
     @property
-    def from_text(self):
+    def send_account_from_text(self):
         return Text(self.driver, prefix=self.locator,
                     xpath="//*[@content-desc='context-tag'][2]/android.widget.TextView").text
 
     @property
-    def to_text(self):
+    def swap_amount_to_text(self):
+        return Text(self.driver, prefix=self.locator,
+                    xpath="//*[@content-desc='context-tag'][2]/android.widget.TextView").text
+
+    @property
+    def account_to_text(self):
         return Text(self.driver, prefix=self.locator,
                     xpath="//*[@content-desc='context-tag'][3]/android.widget.TextView").text
 
@@ -127,6 +161,8 @@ class WalletView(BaseView):
         self.bridge_button = Button(self.driver, accessibility_id='bridge')
         self.send_from_drawer_button = Button(
             self.driver, xpath="//*[@content-desc='send']/*[@content-desc='left-icon-for-action']")
+        self.swap_asset_from_drawer_button = Button( self.driver,
+                                                     xpath="//*[@content-desc='swap']/*[@content-desc='left-icon-for-action']")
         self.copy_address_button = Button(self.driver, accessibility_id='copy-address')
         self.share_address_button = Button(self.driver, accessibility_id='share-account')
         self.remove_account_button = Button(self.driver, accessibility_id='remove-account')
@@ -167,7 +203,13 @@ class WalletView(BaseView):
             self.driver,
             xpath="//*[@content-desc='spender-contract-label']/following-sibling::*[@content-desc='approval-info'][1]")
 
-        # Edit key pair
+        self.max_fees_text =  Text(self.driver, xpath="//*[@text='Max fees']/following-sibling::android.widget.TextView[1]")
+        self.max_slippage_text =  Text(self.driver, xpath="//*[@text='Max slippage']/following-sibling::*[1]")
+        self.est_time_text = Text(self.driver, xpath="//*[@text='Est. time']/following-sibling::*[1]")
+        self.swap_receive_amount_summary_text = Text(self.driver, xpath="//android.widget.TextView[@content-desc='summary-section-receive']/following-sibling::android.view.ViewGroup[1]/android.widget.TextView[1]")
+
+
+        # Edit key pairs
         self.edit_key_pair_button = Button(self.driver, accessibility_id="Edit")
         self.key_pairs_plus_button = Button(self.driver, accessibility_id="standard-title-action")
         self.default_key_pair_container = BaseElement(self.driver,
@@ -251,6 +293,23 @@ class WalletView(BaseView):
         self.continue_button.click()
         self.set_amount(str(amount))
         self.confirm_transaction()
+
+    def swap_asset_from_drawer(self, asset_name: str, amount, network_name: str, decimals_to: int = 4, asset_to: str = 'SNT'):
+        asset_element = self.get_asset(asset_name)
+        asset_element.long_press_without_release()
+        self.swap_asset_from_drawer_button.double_click()
+        self.select_network(network_name)
+        self.set_amount(str(amount))
+        self.max_fees_text.wait_for_visibility_of_element(30)
+        self.confirm_button.click_until_presence_of_element(self.swap_receive_amount_summary_text)
+        expected_amount = self.get_receive_swap_amount(decimals_to)
+        est_time, slippage, fees = self.est_time_text.text, self.max_slippage_text.text, self.max_fees_text.text
+        self.slide_and_confirm_with_password()
+        return { 'receive_amount': expected_amount,
+                 'est_time': est_time,
+                 'est_slippage': slippage,
+                 'max_fees': fees}
+
 
     def add_regular_account(self, account_name: str):
         if not self.add_account_button.is_element_displayed():
@@ -357,14 +416,17 @@ class WalletView(BaseView):
     def round_amount_float(amount, decimals=4):
         return round(float(amount), decimals)
 
-    def wait_for_wallet_balance_to_update(self, expected_amount, asset='Ether', decimals=4):
-        self.just_fyi(f"Checking {asset} balance, expected value: {expected_amount}")
+    def wait_for_wallet_balance_to_update(self, expected_amount, asset='Ether', decimals=4, fiat=False):
+        self.just_fyi(f"Checking {asset} balance, expected value: {expected_amount}, fiat: {fiat}")
 
         start_time = time.time()
 
         while time.time() - start_time < 120:
             self.pull_to_refresh()
-            new_balance = self.round_amount_float(self.get_asset(asset_name=asset).get_amount(), decimals)
+            if fiat is False:
+                new_balance = self.round_amount_float(self.get_asset(asset_name=asset).get_amount(), decimals)
+            else:
+                new_balance = self.get_asset(asset_name=asset).get_fiat_balance()
 
             # Exit early if the balance is updated
             if new_balance == expected_amount:
@@ -373,33 +435,47 @@ class WalletView(BaseView):
             time.sleep(10)  # Wait before retrying
 
         # If the balance is not updated within the timeout, log an error and raise an exception
-        error_message = f"{asset} balance is {new_balance} but expected {expected_amount}"
+        error_message = f"{asset} balance is {new_balance} but expected {expected_amount}, fiat: {fiat}"
         raise TimeoutError(error_message)
 
     def check_last_transaction_in_activity(self, device_time, amount,
+                                           tx_type: Literal['Send', 'Receive', 'Swap'] = 'Send',
                                            asset='ETH',
-                                           to_account='',
-                                           tx_type: Literal['Send', 'Receive'] = 'Send',
                                            from_account_name='Account 1',
+                                           send_to_account='',
+                                           swap_asset_to = 'SNT',
+                                           swap_amount_to = '0.000000000000000000',
                                            network='Status Network'):
         errors = list()
-
-        self.just_fyi("Checking the last transaction in the activity tab")
         current_time = datetime.datetime.strptime(device_time, "%Y-%m-%dT%H:%M:%S%z")
         expected_time = "Today %s" % current_time.strftime('%-I:%M %p')
         possible_times = [expected_time,
                           "Today %s" % (current_time + datetime.timedelta(minutes=1)).strftime('%-I:%M %p')]
-        receiver_variants = [to_account.replace(to_account[5:-3], '...').lower(), 'receiver eth2']
-        activity_element = self.get_activity_element()
+        receiver_variants = [send_to_account.replace(send_to_account[5:-3], '...').lower(), 'receiver eth2']
+        self.just_fyi("Checking the last transaction in the activity tab")
+
         try:
-            checks = {
+            activity_element = self.get_activity_element()
+            common_checks = {
                 "header": (activity_element.header, tx_type),
                 "timestamp": (activity_element.timestamp, possible_times),
                 "amount": (activity_element.amount, f"{amount} {asset}"),
-                "from_text": (activity_element.from_text, from_account_name),
-                "to_text": (activity_element.to_text, receiver_variants),
                 "on_network": (activity_element.on_network, network),
             }
+            if tx_type == 'Send':
+                additional_checks = {
+                    "from_text": (activity_element.send_account_from_text, from_account_name),
+                    "to_text": (activity_element.account_to_text, receiver_variants),
+                }
+            elif tx_type == 'Swap':
+                additional_checks ={
+                    "amount_to": (activity_element.swap_amount_to_text, f"{swap_amount_to} {swap_asset_to}"),
+                    "in_account": (activity_element.account_to_text, from_account_name),
+                }
+            else:
+                additional_checks = {}
+
+            checks = {**common_checks, **additional_checks}
 
             for name, (left, right) in checks.items():
                 if isinstance(right, list):
@@ -421,6 +497,15 @@ class WalletView(BaseView):
             self.close_account_button.click_until_presence_of_element(self.show_qr_code_button)
         return errors
 
-    def get_balance(self, asset='Ether'):
+    def get_balance(self, asset='Ether', fiat=False):
         self.just_fyi("Getting %s amount of the wallet of the sender before transaction" % asset)
-        return self.get_asset(asset_name=asset).get_amount()
+        return self.get_asset(asset_name=asset).get_amount() if fiat is False else self.get_asset(
+            asset_name=asset).get_fiat_balance()
+
+
+    def get_receive_swap_amount(self, decimals=18):
+        self.just_fyi("Getting swap Receive amount for on review page")
+        return self.round_amount_float(self.swap_receive_amount_summary_text.text.split()[0], decimals)
+
+
+

